rllib/agents/pg/pg.py

```diff
@@ -24,6 +50,34 @@ class PGTrainer(Trainer):
     the default `execution_plan()` of `Trainer`.
     """
 
+    def setup(self, config: PartialTrainerConfigDict):
+        super(PGTrainer, self).setup(config)
+        self._train_trainer = None
+        if self.config["framework"] == "torch":
+            config_copy = copy.deepcopy(self.config)
+            self._num_train_workers = config_copy["num_training_workers"]
+            num_cpus_per_training_worker = config_copy["num_cpus_per_training_worker"]
+            num_gpus_per_training_worker = config_copy["num_gpus_per_training_worker"]
+
+            self._train_trainer = train_trainer(
+                "torch",
+                logdir="/tmp/rllib_tmp",
+                num_workers=self._num_train_workers,
+                resources_per_worker={
+                    "CPU": num_cpus_per_training_worker,
+                    "GPU": num_gpus_per_training_worker,
+                },
```

Comment:
I guess you're using CPU only now, but if you want to use GPU you'll need to add something like : `use_gpu=bool(num_gpus_per_training_worker)`